server/src/test/java/org/eclipse/lsp/cobol/usecases/TestTyposOnCompilerDirectives.java

```diff
@@ -43,22 +43,22 @@
           + "024200 PROCEDURE DIVISION .                                             CM1014.2\r\n";
 
   private static final String DATA_ERROR_MESSAGE =
-      "Syntax error on 'DDATA' expected {ADATA, ADV, APOST, ARITH, AWO, BLOCK0, BUF, BUFSIZE, CBLCARD, COBOL2, COBOL3, "
-          + "COMPILE, CPP, CPSM, CURR, DATEPROC, DECK, DIAGTRUNC, DLL, DTR, DYN, DYNAM, EDF, EPILOG, EXP, EXPORTALL, "
-          + "FASTSRT, FEPI, FLAG, FLAGSTD, FSRT, INTDATE, LANG, LEASM, LIB, LIN, LINECOUNT, MARGINS, MDECK, NOADATA, "
-          + "NOADV, NOAWO, NOBLOCK0, NOC, NOCBLCARD, NOCICS, NOCMPR2, NOCOMPILE, NOCPSM, NOCURR, NOCURRENCY, NOD, "
-          + "NODATEPROC, NODBCS, NODEBUG, NODECK, NODIAGTRUNC, NODLL, NODP, NODTR, NODU, NODYN, NODYNAM, NOEDF, "
-          + "NOEPILOG, NOEXIT, NOEXP, NOEXPORTALL, NOF, NOFASTSRT, NOFEPI, NOFLAG, NOFLAGMIG, NOFLAGSTD, NOFSRT, "
-          + "NOGRAPHIC, NOLENGTH, NOLIB, NOLINKAGE, NOLIST, NOMAP, NOMD, NOMDECK, NONAME, NONUM, NONUMBER, NOOBJ, "
-          + "NOOBJECT, NOOFF, NOOFFSET, NOOPSEQUENCE, NOOPT, NOOPTIMIZE, NOOPTIONS, NOP, NOPROLOG, NORENT, NOS, NOSEQ, "
-          + "NOSOURCE, NOSPIE, NOSQL, NOSQLC, NOSQLCCSID, NOSSR, NOSSRANGE, NOSTDTRUNC, NOSEQUENCE, NOTERM, "
-          + "NOTERMINAL, NOTEST, NOTHREAD, NOVBREF, NOWD, NOWORD, NOX, NOXREF, NOZWB, NSEQ, NSYMBOL, NUM, NUMPROC, "
-          + "OBJ, OFF, OFFSET, OPMARGINS, OPSEQUENCE, OPT, OPTFILE, OUTDD, PGMN, PGMNAME, PROLOG, RENT, RES, RMODE, "
-          + "SEQ, SEQUENCE, SPIE, SQLC, SQLCCSID, SSR, SSRANGE, SYSEIB, TERM, TRUNC, VBREF, WORD, XOPTS, XMLPARSE, "
-          + "XREF, YEARWINDOW, ZWB, CODEPAGE, NODE, CICS, CURRENCY, DATA, DBCS, DEBUG, DUMP, LANGUAGE, LENGTH, "
-          + "LINKAGE, LIST, NAME, NATLANG, NODUMP, NUMBER, OBJECT, OPTIMIZE, OPTIONS, OUT, QUOTE, SQL, EXIT, GDS, "
-          + "GRAPHIC, MAP, SIZE, SOURCE, SPACE, TERMINAL, TEST, THREAD, C, D, F, Q, S, X, AR, CP, DP, DU, LC, MD, "
-          + "NS, OP, SP, SZ, WD, XP, YW, ','}";
+      "Syntax error on 'DDATA' expected {ADATA, ADV, APOST, ARITH, AWO, BLOCK0, BUF, BUFSIZE, CBLCARD, COBOL2, "
```

Comment:
How is about sorting those keywords somehow?

Reply:
Well, I would first separate the CICS parser. It will definitely change error messages like this, so we'll see how to improve them.